modules/imaging_uploader/jsx/upload_form.js

```diff
@@ -0,0 +1,141 @@
+class UploadForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formData: {},
+      form: this.props.form,
+      uploadProgress: -1
+    };
+
+    this.onFormChange = this.onFormChange.bind(this);
+    this.uploadFile = this.uploadFile.bind(this);
+  }
+
+  componentDidMount() {
+    let form = this.props.form;
```

Comment:
I can, but I still need to have a deep copy of props in upload form. So I moved it from `<ImagingUploader/>` to `<UploadForm/>` constructor per your suggestion